dev-tools/gen-target-info/src/main.rs

```diff
@@ -1,52 +1,73 @@
-use gen_target_info::{get_target_specs_from_json, write_target_tuple_mapping, RustcTargetSpecs};
-use std::{collections::BTreeMap, fs::File, io::Write as _};
+use std::io::Write as _;
+use std::{fs::File, io::BufRead};
+
+use gen_target_info::{
+    get_target_spec_from_msrv, get_target_specs_from_json, get_targets_msrv, RustcTargetSpecs,
+};
 
 const PRELUDE: &str = r#"//! This file is generated code. Please edit the generator
 //! in dev-tools/gen-target-info if you need to make changes.
 
 "#;
 
-fn generate_riscv_arch_mapping(f: &mut File, target_specs: &RustcTargetSpecs) {
-    let riscv_target_mapping = target_specs
-        .0
-        .iter()
-        .filter_map(|(target, target_spec)| {
-            let arch = target.split_once('-').unwrap().0;
-            (arch.contains("riscv") && arch != target_spec.arch)
-                .then_some((arch, &*target_spec.arch))
-        })
-        .collect::<BTreeMap<_, _>>();
-    write_target_tuple_mapping(f, "RISCV_ARCH_MAPPING", &riscv_target_mapping);
-}
+fn generate_target_mapping(f: &mut File, target_specs: &RustcTargetSpecs) -> std::io::Result<()> {
+    writeln!(f, "use super::Target;")?;
+    writeln!(f)?;
+    writeln!(
+        f,
+        "pub(crate) fn get(target_triple: &str) -> Option<Target> {{"
+    )?;
```

Comment:
I think write_str is more readable, since you can use `{` instead of `{{`, which is really hard to read

```suggestion
    f.write_str("pub(crate) fn get(target_triple: &str) -> Option<Target> {"
    )?;
```

Reply:
I have decided to leave it in for `Target {{`, since I think it makes it clearer to read that the formatting lines up.